tests/conftest.py

```diff
@@ -0,0 +1,21 @@
+import os
+import platform
+import sys
+
+import matplotlib
+
+# Force matplotlib to use a non-interactive backend and disable showing plots
+os.environ.setdefault("MPLBACKEND", "Agg")
+matplotlib.use("Agg", force=True)
+matplotlib.interactive(False)
+
+
+def pytest_sessionfinish(session, exitstatus):
```

Comment:
This is necessary because for Python 3.12 with Mac Silicon -- there's some weird issue with the torch DataLoader's workers not closing which will cause pytest to hang forever.